sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -187,37 +137,36 @@ func (r *genericRequester[ReqT, RespT]) RequestCertificates(ctx context.Context)
 		}
 	}
 
+	if r.ongoingRequest.Load() {
+		return nil, errors.New("concurrent requests are not supported.")
+	}
+	r.ongoingRequest.Store(true)
```

Comment:
Isn't there a race here? It seems two concurrent `RequestCertificates` calls could see no ongoing request, and then both would store a `true` and then proceed (with one overwriting the request id of the other).

Reply:
Yes there is, thanks for spotting this!